include/envoy/network/transport_socket.h

```diff
@@ -109,9 +115,11 @@ class TransportSocket {
 
   /**
    * @param buffer supplies the buffer to write from
+   * @param end_stream supplies whether this is the end of the stream. If true and all
+   *        data in buffer is written, the connection will be half-closed.
```

Comment:
I decided to ASSERT that this doesn't happen (and document it as such). Let me know if you think it would be better to translate it; my feeling is it shouldn't happen.  If half-close isn't enabled, just call close().